packages/alea-frontend/pages/api/university-admin/semester-info/get-holidays-info.ts

```diff
@@ -0,0 +1,31 @@
+import { Action, ResourceName } from '@stex-react/utils';
+import { getUserIdIfAuthorizedOrSetError } from '../../access-control/resource-utils';
+import { executeQuery, checkIfGetOrSetError} from '../../comment-utils';
+
+export default async function handler(req, res) {
+  if (!checkIfGetOrSetError(req, res)) return;
+
+  const userId = await getUserIdIfAuthorizedOrSetError(req,res,ResourceName.UNIVERSITY_SEMESTER_DATA,Action.MUTATE,{universityId: req.query.universityId});

```

Comment:
Viewing the semester-info/holiday info doesn't need authorization check. Please check which APIs need this and which ones dont.